libcloud/compute/drivers/runabove.py

```diff
@@ -52,13 +39,20 @@ class RunAboveNodeDriver(NodeDriver):
     api_name = 'runabove'
 
     NODE_STATE_MAP = OpenStackNodeDriver.NODE_STATE_MAP
+    VOLUME_STATE_MAP = OpenStackNodeDriver.VOLUME_STATE_MAP
```

Comment:
I've missed this when originally reviewing the PR - it looks like their offering is based on OpenStack.

Any particular reason why you didn't inherit from the base OpenStack driver (that's what a lot of other drivers for providers which are based on the OpenStack do).

The auth is not using Keystone, but I quickly glanced over other methods and at least some of them look similar to the ones implemented in the OpenStack driver :)